tensorflow/core/kernels/dml_softmax_ops.cc

```diff
@@ -27,14 +28,18 @@ class SoftmaxXentWithLogitsShapeHelper : public ShapeHelper {
   std::vector<TensorShape> GetOutputShapes(
       OpKernelContext* ctx,
       const InitializationHelper* initialization_helper) const override {
-    const Tensor& logits = ctx->input(0);
-    const Tensor& labels = ctx->input(1);
-    // logits must have the same shape as labels
-    CHECK(logits.shape() == labels.shape());
+    const Tensor& logits_in = ctx->input(0);
+    const Tensor& labels_in = ctx->input(1);
 
-    TensorShape outputShape({logits.dim_size(0), 1});
+    BCast bcast(BCast::FromShape(logits_in.shape()),

```

Comment:
You have the same logic in the init helper. Since you need to broadcast in the init helper anyway, the output shapes should probably be computed in the init helper, and `GetOutputShapes()` should only be a thin wrapper than fetches them and returns them.